.github/workflows/build-and-test.yml

```diff
@@ -133,7 +133,7 @@ jobs:
   unittest:
     strategy:
       matrix:
-        go-version: ["1.20", 1.19] # 1.20 needs quotes otherwise it's interpreted as 1.2
+        go-version: ["1.21", "1.20"] # 1.20 needs quotes otherwise it's interpreted as 1.2
```

Comment:
Should this also use the `~` convention to force min version? Aligning across actions could make search and replace easier in the future.

Reply:
Happy to do that, but I am worried that `~` is not a supported character on cache keys and since we use this variable here https://github.com/open-telemetry/opentelemetry-collector/blob/929cce9f18fb3014385da1d54fc4137efe434aec/.github/workflows/build-and-test.yml#L158 it would break things, so I would prefer to work on this on a separate PR.